client.go

```diff
@@ -326,7 +326,7 @@ func (c *client) recv(l *net.UDPConn, msgCh chan *dns.Msg) {
 		return
 	}
 	buf := make([]byte, 65536)
-	for !c.closed {
```

Comment:
if you're eliminating the close variable you could simply get rid of it, but I don't think the mutex is really required.